src/distributions/other.rs

```diff
@@ -145,6 +147,39 @@ impl Distribution<bool> for Standard {
     }
 }
 
+/// Requires nightly Rust and the [`simd_support`] feature
+///
+/// Note that on some hardware like x86/64 mask operations like [`_mm_blendv_epi8`]
+/// only care about a single bit. This means that you could use uniform random bits
+/// directly:
+///
+/// ```ignore
+/// // this may be faster...
+/// let x = unsafe { _mm_blendv_epi8(a.into(), b.into(), rng.gen::<__m128i>()) };
+///
+/// // ...than this
+/// let x = rng.gen::<mask8x16>().select(b, a);
+/// ```
+///
+/// Since most bits are unused you could also generate only as many bits as you need.
+///
+/// [`_mm_blendv_epi8`]: https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_blendv_epi8&ig_expand=514/
+/// [`simd_support`]: https://github.com/rust-random/rand#crate-features
+#[cfg(feature = "simd_support")]
+impl<T, const LANES: usize> Distribution<Mask<T, LANES>> for Standard
+where
+    T: MaskElement + PartialOrd + SimdElement<Mask = T> + Default,
+    LaneCount<LANES>: SupportedLaneCount,
+    Standard: Distribution<Simd<T, LANES>>,
+{
+    #[inline]
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Mask<T, LANES> {
+        // `MaskElement` must be a signed integer, so this is equivalent
+        // to the scalar `i32 < 0` method
+        rng.gen().lanes_lt(Simd::default())
+    }
+}
```

Comment:
From what I understand this is correct, but not necessarily the most efficient: only the high bit of each lane generated by `rng.gen()` is used. I'm not sure if this is actually worth worrying about however.

I'm also just a little surprised it type-checks: it relies on only one type supporting `lanes_lt(..) -> Mask<T, LANES>`. Better to clarify with `rng.gen::<T>().lates_lt(T::default())`?

Reply:
The inefficiency can actually be worse in scalar. `mask8x16` wastes 7 bits per lane while `bool` wastes 31, `mask64x2` though wastes 63. I'll look into more efficient methods but might just point readers to it in the docs